src/hooks/posters.ts

```diff
@@ -71,24 +72,55 @@ export const usePosters = (posterHallId: string) => {
   const fuseVenues = useMemo(
     () =>
       new Fuse(filteredPosterVenues, {
-        keys: ["poster.title", "poster.authorName", "poster.categories"],
+        keys: [
+          "name",
+          "poster.title",
+          "poster.authorName",
+          "poster.categories",
+        ],
+        threshold: 0.2, // 0.1 seems to be exact, default 0.6: brings too distant if anyhow related hits
+        ignoreLocation: true, // default False: True - to search ignoring location of the words.
+        findAllMatches: true,
       }),
     [filteredPosterVenues]
   );
 
   const searchedPosterVenues = useMemo(() => {
-    if (!searchQuery)
-      return filteredPosterVenues.slice(0, displayedPostersCount);
+    const normalizedSearchQuery = searchQuery.trim();
+
+    if (!normalizedSearchQuery) return filteredPosterVenues;
+
+    const matchEntries = breakStringWithQuotesBySpaces(normalizedSearchQuery);
+
+    if (!matchEntries) return filteredPosterVenues;
 
     return fuseVenues
-      .search(searchQuery)
-      .slice(0, displayedPostersCount)
+      .search({
+        $and: matchEntries.map((x) => ({
+          $or: [
+            { name: x },
+            { "poster.title": x },
+            { "poster.authorName": x },
+            { "poster.categories": x },
+            // @debt Rewrite the search query to match the $and type
+          ] as Record<string, string>[],
```

Comment:
Why are we explicitly forcing the type here with `as`? Surely there's a better solution?

Reply:
This seems to be hiding an error: 
![image](https://user-images.githubusercontent.com/753891/119936786-0bacb180-bfcd-11eb-9254-7ca8db0bd56a.png)

The last line seems to imply that `undefined` is not an allowable value here, yet I believe that's the only reason why the `as` type override is needed here.

Reply:
Actually.. that doesn't seem to fully solve it.. 🔍 👀

Reply:
This seems to solve it:

```patch
diff --git a/src/hooks/posters.ts b/src/hooks/posters.ts
index 9714f48a..853270cf 100644
--- a/src/hooks/posters.ts
+++ b/src/hooks/posters.ts
@@ -90,21 +90,26 @@ export const usePosters = (posterHallId: string) => {
 
     if (!normalizedSearchQuery) return filteredPosterVenues;
 
-    const matchEntries = breakStringWithQuotesBySpaces(normalizedSearchQuery);
+    const matchEntries: string[] =
+      breakStringWithQuotesBySpaces(normalizedSearchQuery) ?? [];
 
     if (!matchEntries) return filteredPosterVenues;
 
+    // @debt Rewrite the search query to match the $and type
     return fuseVenues
       .search({
-        $and: matchEntries.map((x) => ({
-          $or: [
+        $and: matchEntries.map((x: string) => {
+          const orFields: Fuse.Expression[] = [
             { name: x },
             { "poster.title": x },
             { "poster.authorName": x },
             { "poster.categories": x },
-            // @debt Rewrite the search query to match the $and type
-          ] as Record<string, string>[],
-        })),
+          ];
+
+          return {
+            $or: orFields,
+          };
+        }),
       })
       .map((fuseSearchItem) => fuseSearchItem.item);
   }, [searchQuery, fuseVenues, filteredPosterVenues]);
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1460/commits/56686771a3a97eb5d0a537bb191dfad4754d362f